runsc/cmd/exec.go

```diff
@@ -140,6 +144,35 @@ func (ex *Exec) Execute(_ context.Context, f *flag.FlagSet, args ...any) subcomm
 		log.Infof("Using exec capabilities from container: %+v", e.Capabilities)
 	}
 
+	// Create the file descriptor map for the process in the container.
+	fdMap := map[int]*os.File{
+		0: os.Stdin,
+		1: os.Stdout,
+		2: os.Stderr,
+	}
+
+	// Add custom file descriptors to the map.
+	var files []*os.File
+	for _, fd := range ex.passFDs {
+		file := os.NewFile(uintptr(fd), "")
+		if file == nil {
+			util.Fatalf("failed to create file from file descriptor %d", fd)
+		}
+		fdMap[int(file.Fd())] = file
```

Comment:
This means that the files passed into the sandbox will have the same FD inside and outside the sandbox, which is a little bit inflexible.

Should the `exec` and `run` commands support the the `--custom-fds` mapping like boot does?  Where we specify the mapping from host to guest fd numbers?

Reply:
I have changed the `--pass-fd` types to the `fdMapping` type in c66f9f08bdae58f94ff2540a9d7b2acaa799bb9b and added handling for single integers, which is a shortcut for no remapping.